src/code/audio/code_8019AF00.c

```diff
@@ -1953,17 +1957,152 @@ const char sAudioOcarinaUnusedText7[] = "check is over!!! %d %d %d\n";
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019CF78.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019CF9C.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/AudioOcarina_GetPlaybackStaff.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019CFA8.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D134.s")
+/**
+ * Tests to see if the notes from songIdx contain identical notes
+ * within its song to any of the reserved songIdx from 0 up to maxSongIdx
+ */
+s32 AudioOcarina_TerminaWallValidateNotes(u8 songIdx, u8 maxSongIdx) {
+    u8 curSongIdx;
+    u8 j;
+    u8 k;
+
+    // loop through all possible songs up to maxSongIdx
+    for (curSongIdx = 0; curSongIdx < maxSongIdx; curSongIdx++) {
+        // check to see if the song is reserved or not
+        if (sIsOcarinaSongReserved[curSongIdx]) {
+            // starting index to test the song
+            for (j = 0; j < (9 - gOcarinaSongButtons[curSongIdx].numButtons); j++) {
+                // loop through each note in the song
+                for (k = 0;
+                     (k < gOcarinaSongButtons[curSongIdx].numButtons) && ((k + j) < 8) &&
+                     (gOcarinaSongButtons[curSongIdx].buttonIdx[k] == gOcarinaSongButtons[songIdx].buttonIdx[(k + j)]);
+                     k++) {
+                    continue;
+                }
+
+                if (k == gOcarinaSongButtons[curSongIdx].numButtons) {
+                    // failure: songIdx is identical to curSongIdx.
+                    return -1;
+                }
+            }
+        }
+    }
+
+    // success: notes are accepted and used
+    return 0;
+}
+
+/**
+ * Generates the notes displayed on the Termina Field wall of musical notes
+ * Song generation iterates back and forth from 8 random notes to a song from Ocarina of Time (OoT).
```

Comment:
```suggestion
 * Song generation loop alternates between 8 random notes and a random song from Ocarina of Time (OoT).
```
seems clearer to me.